### PR TITLE
chore: add changelog for 12.10.0

### PR DESCRIPTION
**Pull request by astone123**
This PR adds the changelog for Cypress 12.10.0 to the documentation https://github.com/cypress-io/cypress/pull/26482

### CODE REVIEW

**Review by AtofStryker — APPROVED** [review]

**Comment by AtofStryker** [comment]
looks good. Likely just need to run the `lint:fix` npm script locally and push back up

File: docs/guides/references/changelog.mdx [hunk 1]
```diff
@@ -2,6 +2,65 @@
 title: Changelog
 ---
 
+## 12.10.0
+
+_Released 04/11/2023_
```
Comment (emilyrohrbough) [inline docs/guides/references/changelog.mdx:7]:
```suggestion
_Released 04/12/2023_
```

File: docs/guides/references/changelog.mdx [hunk 2]
```diff
@@ -2,6 +2,65 @@
...
+- Fix an edge case in Component Testing where a custom `baseUrl` in
+  `tsconfig.json` for Next.js 13.2.0+ is not respected. This was partially fixed
+  in [#26005](https://github.com/cypress-io/cypress/pull/26005), but an edge
+  case was missed. Fixes
+  [#25951](https://github.com/cypress-io/cypress/issues/25951).
+- Correctly detect and resolve dependencies when configuring Component Testing
+  in projects using Yarn's
+  [Plug'n'Play feature](https://yarnpkg.com/features/pnp). Fixes
+  [#25960](https://github.com/cypress-io/cypress/issues/25960).
+- Fixed an issue where `click` events fired on `.type('{enter}')` did not
```
Comment (emilyrohrbough) [inline docs/guides/references/changelog.mdx:42]:
```suggestion
- Fixed an issue where `click` events fired on [`.type('{enter}')`](/api/commands/type) did not
```

File: docs/guides/references/changelog.mdx (hunk 1 above)
Comment (mjhenkes) [reply docs/guides/references/changelog.mdx:7]:
```suggestion
_Released 04/17/2023_
```

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**